src/orange/OrangeData.hh

```diff
@@ -146,6 +149,41 @@ struct RaggedRightIndexerData
     }
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Data for a single Bounding Inverval Hierarchy node
+ *
+ * Leaf nodes must contain exactly one bounding box
+ */
+struct BIHNode
+{
+    enum Edge : size_type
+    {
+        left = 0,
+        right = 1
+    };
+
+    // inner node only
+    Array<BIHNodeId, 2> children;
+    Array<float, 2> partitions;
+
+    // leaf only
+    ItemRange<LocalVolumeId> vol_ids;
+
+    CELER_FUNCTION bool is_inner() const
+    {
+        return this->children[Edge::left] && this->children[Edge::right];
+    }
+
+    CELER_FUNCTION bool is_leaf() const { return !vol_ids.empty(); }
+
+    //! True if either a valid inner or leaf node
+    explicit CELER_FUNCTION operator bool() const
+    {
+        return this->is_inner() ^ this->is_leaf();
```

Comment:
Be careful accidentally using bitwise operators when you mean boolean operators:
```suggestion
        return this->is_inner() != this->is_leaf();
```